Util/Tokenization/src/main/java/org/tribuo/util/tokens/impl/wordpiece/WordpieceTokenizer.java

```diff
@@ -0,0 +1,205 @@
+package org.tribuo.util.tokens.impl.wordpiece;
+
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.tribuo.util.tokens.Token;
+import org.tribuo.util.tokens.Token.TokenType;
+import org.tribuo.util.tokens.Tokenizer;
+
+import com.oracle.labs.mlrg.olcut.config.Config;
+import com.oracle.labs.mlrg.olcut.provenance.ConfiguredObjectProvenance;
+import com.oracle.labs.mlrg.olcut.provenance.impl.ConfiguredObjectProvenanceImpl;
+
+public class WordpieceTokenizer implements Tokenizer {
+
+    @Config
+    private Wordpiece wordpiece;
+    @Config
+    private boolean toLowerCase = true;
+    @Config
+    private Tokenizer tokenizer = new WordpiecePreprocessTokenizer();
+    @Config
+    private boolean stripAccents = true;
+    @Config
+    private Set<String> neverSplitTokens = Collections.emptySet();
+
+    // internal state member variables
+    private boolean reset;
+
+    private Token currentToken;
+
+    private List<Token> currentWordpieceTokens = new ArrayList<>();
+
+    private int currentWordpieceIndex;
+
+    // for OLCUT
+    @SuppressWarnings("unused")
+    private WordpieceTokenizer() {
+    }
+
+    /**
+     * @param wordpiece        an instance of {@link Wordpiece}
+     * @param tokenizer        Wordpiece is run on the tokens generated by the
+     *                         tokenizer provided here.
+     * @param toLowerCase      determines whether or not to lowercase each token
+     *                         before running Wordpiece on it
+     * @param stripAccents     determines whether or not to strip out accents from
+     *                         each token before running Wordpiece on it
+     * @param neverSplitTokens a set of token values that we will not apply
+     *                         Wordpiece to. This tokenizer does not coordinate with
+     *                         the passed in tokenizer to make sure that it will not
+     *                         split up the values in neverSplitTokens. You must
+     *                         provide a tokenizer that will not split them up.
+     */
+    public WordpieceTokenizer(Wordpiece wordpiece, Tokenizer tokenizer, boolean toLowerCase, boolean stripAccents,
+            Set<String> neverSplit) {
+        this.wordpiece = wordpiece;
+        this.tokenizer = tokenizer;
+        this.toLowerCase = toLowerCase;
+        this.stripAccents = stripAccents;
+        this.neverSplitTokens = neverSplit;
+    }
+
+    @Override
+    public ConfiguredObjectProvenance getProvenance() {
+        return new ConfiguredObjectProvenanceImpl(this, "Tokenizer");
+    }
+
+    public static List<String> whitespaceTokenize(String text) {
```

Comment:
Do we need this given it's on `Wordpiece` as well?

Reply:
removed!  from both, actually.